modules/imap/workers/messages_list.php

```diff
@@ -3,75 +3,81 @@
-$data = json_decode($input, true);
+    $input = trim(fgets(STDIN));
+    $data = json_decode($input, true);
 
-$index = $data['index'];
-$search = $data['search'];
-$dataSource = $data['dataSource'];
-$cache = unserialize($data['cache']);
-$cache->reconnect();
-$session = unserialize($data['session']);
-$user_config = unserialize($data['config']);
+    $index = $data['index'];
+    $search = $data['search'];
+    $dataSource = $data['dataSource'];
+    $cache = unserialize($data['cache']);
+    $cache->reconnect();
+    $session = unserialize($data['session']);
+    $user_config = unserialize($data['config']);
 
-$filter = $search['filter'];
-$sort = $search['sort'];
-$reverse = $search['reverse'];
-$searchTerms = $search['terms'];
-$limit = $search['limit'];
-$offsets = $search['offsets'];
-$listPage = $search['listPage'];
+    $filter = $search['filter'];
+    $sort = $search['sort'];
+    $reverse = $search['reverse'];
+    $searchTerms = $search['terms'];
+    $limit = $search['limit'];
+    $offsets = $search['offsets'];
+    $listPage = $search['listPage'];
 
-$offset = $search['defaultOffset'];
-if ($offsets && $listPage > 1) {
-    if (isset($offsets[$index]) && (int) $offsets[$index] > 0) {
-        $offset = (int) $offsets[$index] * ($listPage - 1);
+    $offset = $search['defaultOffset'];
+    if ($offsets && $listPage > 1) {
+        if (isset($offsets[$index]) && (int) $offsets[$index] > 0) {
+            $offset = (int) $offsets[$index] * ($listPage - 1);
+        }
     }
-}
 
-Hm_IMAP_List::init($user_config, $session);
-$mailbox = Hm_IMAP_List::get_connected_mailbox($dataSource['id'], $cache);
-if ($mailbox && $mailbox->authed()) {
-    $search['offset'] = $offset;
-    $result = getMessagesList($mailbox, $dataSource, $search);
+    Hm_IMAP_List::init($user_config, $session);
+    $mailbox = Hm_IMAP_List::get_connected_mailbox($dataSource['id'], $cache);
+    if ($mailbox && $mailbox->authed()) {
+        $search['offset'] = $offset;
+        $result = getMessagesList($mailbox, $dataSource, $search);
 
-    echo json_encode([
-        'uids' => $result['uids'],
-        'status' => $result['status'],
-        'messages' => $result['messages'],
-        'total' => $result['total'],
-        'dataSource' => $result['dataSource'],
-        'folder' => $result['folder']
-    ]);
-} else {
-    echo json_encode(['error' => 'Failed to establish connection to the mailbox.']);
+        echo json_encode([
+            'uids' => $result['uids'],
+            'status' => $result['status'],
+            'messages' => $result['messages'],
+            'total' => $result['total'],
+            'dataSource' => $result['dataSource'],
+            'folder' => $result['folder']
+        ]);
+    } else {
+        echo json_encode(['error' => 'Failed to establish connection to the mailbox.']);
+        exit;
+    }
+} catch (\Exception $e) {
+    Hm_Functions::error_log(sprintf(' error processing mailbox' .$e->getMessage() ));
```

Comment:
Isn't it better to write to stderr any error messages and let the calling scripts decide how to log them? Otherwise, we have logging patterns on too many locations and it is harder to maintain.

Reply:
Btw, even if you don't add a big try/catch block, exceptions thrown in the related code should end up in the stderr stream of the process and thus can be catched in the calling code.